docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/ApproximatedKernelMap.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+using Microsoft.ML.Transforms;
+
+namespace Samples.Dynamic
+{
+    public static class ApproximatedKernelMap
+    {
+        // Transform feature vector to another non-linear space. See https://people.eecs.berkeley.edu/~brecht/papers/07.rah.rec.nips.pdf.
+        public static void Example()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var mlContext = new MLContext();
+            var samples = new List<DataPoint>()
+            {
+                new DataPoint(){ Features = new float[7] { 1, 1, 0, 0, 1, 0, 1} },
+                new DataPoint(){ Features = new float[7] { 0, 0, 1, 0, 0, 1, 1} },
+                new DataPoint(){ Features = new float[7] {-1, 1, 0,-1,-1, 0,-1} },
+                new DataPoint(){ Features = new float[7] { 0,-1, 0, 1, 0,-1,-1} }
+            };
+            // Convert training data to IDataView, the general data type used in ML.NET.
+            var data = mlContext.Data.LoadFromEnumerable(samples);
+            // ApproximatedKernel map takes data and maps it's to a random low-dimensional space.
+            var approximation = mlContext.Transforms.ApproximatedKernelMap("Features", rank: 4, generator: new GaussianKernel(gamma: 0.7f), seed: 1);
+
+            // Now we can transform the data and look at the output to confirm the behavior of the estimator.
+            // This operation doesn't actually evaluate data until we read the data below.
+            var tansformer = approximation.Fit(data);
+            var transformedData = tansformer.Transform(data);
+
+            var column = transformedData.GetColumn<float[]>("Features").ToArray();
+            foreach (var row in column)
+                Console.WriteLine(string.Join(", ", row.Select(x => x.ToString("f4"))));
+            // Expected output:
+            // -0.0119, 0.5867, 0.4942,  0.7041
```

Comment:
>  [](start = 14, length = 1)

Space Space

Reply:
I prefer to align numbers, so one space was taken by minus sign.

---
In reply to: [273740487](https://github.com/dotnet/machinelearning/pull/3232#discussion_r273740487) [](ancestors = 273740487)